MCT-4697: Button Redesign

erinrep opened the pull request:
#### Fixes [MCT-4697](https://jira.cms.gov/browse/MCT-4697)
Redesign for the various buttons

#### Updated doc page
![Screenshot_2021-01-11 Button - Medicare gov Design System](https://user-images.githubusercontent.com/4550709/104205173-54cbe280-53eb-11eb-8a59-eb7780af4288.png)





### Change guide for the release notes
![Screen Shot 2020-12-14 at 8 47 57 AM](https://user-images.githubusercontent.com/4550709/103580808-1341b180-4e98-11eb-8278-49412e3afb71.png)

babsmiles reviewed (approved):
This looks really great and the release notes you made are going to help so much. Thank you!

bernardwang commented on docs/src/pages/components/Button/Button.example.tsx:
```diff
@@ -5,201 +5,15 @@ import { Button } from "@cmsgov/design-system";
-          </td>
-        </tr>
-      </table>
+      <span className="ds-u-margin-right--1">
```
@erinrep What do you think about a follow up PR to make the React and HTML examples identical? This way folks know what props to use to achieve a specific style?

bernardwang commented on docs/src/pages/components/Button/button.example.html:
```diff
@@ -93,17 +93,39 @@ <h6 class="preview__label">Transparent button</h6>
 
 <h6 class="preview__label">Inverse buttons</h6>
 <div class="example--inverse">
+  <div class="ds-u-margin-bottom--1">
```
Maybe we can add some labels here as well, like "Inverse Primary", "Inverse Default", "Inverse Transparent" and "Inverse Landing Page"

bernardwang commented on docs/src/pages/components/Button/button.example.html:
```diff
@@ -163,4 +185,30 @@ <h6 class="preview__label">Inverse buttons</h6>
   </div>
-</div>
+  <div>
+    <button type="button" class="ds-c-button ds-c-button--primary ds-c-button--inverse ds-c-button--big">
```
This is a minor thing, but since we dont include all the states for the non inverse landing page button, can we also remove the states for the inverse landing page button to keep things consistent?

bernardwang commented on src/styles/components/_override.buttons.scss:
```diff
@@ -81,44 +111,51 @@
-  color: $color-primary;
+    &:focus,
+    &.ds-c-button--focus {
+      border-color: rgba($color-gray-dark, 0.8);
```
Should the inverse primary focus state have any additional styles? In the normal focus state the color and border color stay the same

erinrep replied:
That is a good question. I modeled it after the main DS where the inverse primary focus state's color and border match the hover state's color and border but the non-inverse one matches the normal state, so maybe we want to keep with that. @babsmiles what are your thoughts?

bernardwang commented on src/styles/components/_override.buttons.scss:
```diff
@@ -81,44 +111,51 @@
-  }
+    &:active,
+    &.ds-c-button--active {
+      border-color: $color-gray-dark;
```
I think this should also be using rgba
`border-color: rgba($color-gray-dark, 0.6);`

bernardwang reviewed (approved):
There were some minor SCSS issue, but aside from that this looks really great! We can also have a followup PR to address the other documentation related comments

erinrep commented:
@babsmiles @bernardwang @nichia I updated the documentation page to incorporate Bernard's suggestions:
- labels for the inverse buttons
- omit the landing page inverse buttons states
- added additional React examples
What do you think of those changes? (I updated the screenshot in the description)
